Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -460,6 +524,59 @@ void HTMLAttachmentElement::setFile(RefPtr<File>&& file, UpdateDisplayAttributes
     invalidateRendering();
 }
 
+// FIXME: Remove after rdar://99228361 is fixed.
+class AttachmentEvent {
+public:
+    const void* attachment() const { return m_attachment; }
+    WTF::MonotonicTime time() const { return m_time; }
+    StackTrace& stackTrace() const { return *m_stackTrace; }
+
+    void capture(const HTMLAttachmentElement& a, WTF::MonotonicTime t = WTF::MonotonicTime::now())
+    {
+        m_attachment = &a;
+        ASSERT(!!t);
+        m_time = t;
+        m_stackTrace = StackTrace::captureStackTrace(64);
+    }
+
+    void reset()
+    {
+        m_attachment = nullptr;
+        m_time = WTF::MonotonicTime();
+        m_stackTrace = nullptr;
+    }
+
+    explicit operator bool() const
+    {
+        ASSERT(!m_attachment == !m_time && !m_attachment == !m_stackTrace);
+        return !!m_attachment;
+    }
+
+private:
+    const void* m_attachment = nullptr;
+    WTF::MonotonicTime m_time;
+    std::unique_ptr<StackTrace> m_stackTrace;
+};
+
+static AttachmentEvent& lastInsertionInDoc()
```

Comment:
This should be `Document` rather than `Doc`. Other instances throughout the patch should also be updated.

(https://webkit.org/code-style-guidelines/#names-full-words)